Azure Function Maven Plugin fails to package

### Plugin name and version

`azure-functions-maven-plugin` version `1.3.4`

### Plugin configuration in your `pom.xml`

```xml
<plugin>
    <groupId>com.microsoft.azure</groupId>
    <artifactId>azure-functions-maven-plugin</artifactId>
    <version>${azure.functions.maven.plugin.version}</version>
</plugin>
```

### Expected behavior

Should package. The equivalent Gradle plugin doesn't exhibit the problem.

### Actual behavior

Fails to package with:

```
[ERROR] Failed to execute goal com.microsoft.azure:azure-functions-maven-plugin:1.3.4:package (package-functions) on project azure-functions-example: Can't resolve member named invoke for class example.Function -> [Help 1]
[ERROR]
```

The stacktrace is:

```
Caused by: org.reflections.ReflectionsException: Can't resolve member named invoke for class example.Function
	at org.reflections.util.Utils.getMemberFromDescriptor(Utils.java:94)
	at org.reflections.util.Utils.getMethodsFromDescriptors(Utils.java:101)
	at org.reflections.Reflections.getMethodsAnnotatedWith(Reflections.java:482)
	at com.microsoft.azure.maven.function.handlers.AnnotationHandlerImpl.findFunctions(AnnotationHandlerImpl.java:50)
	at com.microsoft.azure.maven.function.PackageMojo.findAnnotatedMethods(PackageMojo.java:135)
	at com.microsoft.azure.maven.function.PackageMojo.doExecute(PackageMojo.java:87)
	at com.microsoft.azure.maven.AbstractAzureMojo.execute(AbstractAzureMojo.java:309)
```


### Steps to reproduce the problem

```
git clone git@github.com:micronaut-projects/micronaut-azure.git
cd examples/azure-functions-example
./mvnw clean package azure-functions:run
```

Note the error. Now try:

```
../gradlew clean azureFunctionsRun
```

Note that with Gradle the problem doesn't exist.

--- a/azure-functions-maven-plugin/src/main/java/com/microsoft/azure/maven/function/PackageMojo.java
+++ b/azure-functions-maven-plugin/src/main/java/com/microsoft/azure/maven/function/PackageMojo.java
@@ -59,7 +59,7 @@ import java.util.Set;
  * Generate configuration files (host.json, function.json etc.) and copy JARs to staging directory.
  */
 @Mojo(name = "package", defaultPhase = LifecyclePhase.PACKAGE,
-        requiresDependencyResolution = ResolutionScope.COMPILE)
+        requiresDependencyResolution = ResolutionScope.RUNTIME)
 public class PackageMojo extends AbstractFunctionMojo {
     public static final String SEARCH_FUNCTIONS = "Step 1 of 7: Searching for Azure Functions entry points";
     public static final String FOUND_FUNCTIONS = " Azure Functions entry point(s) found.";
@@ -172,13 +172,13 @@ public class PackageMojo extends AbstractFunctionMojo {
      */
     protected List<URL> getDependencyArtifactUrls() {
         final List<URL> urlList = new ArrayList<>();
-        final List<String> compileClasspathElements = new ArrayList<>();
+        final List<String> runtimeClasspathElements = new ArrayList<>();
         try {
-            compileClasspathElements.addAll(this.getProject().getCompileClasspathElements());
+            runtimeClasspathElements.addAll(this.getProject().getRuntimeClasspathElements());
         } catch (DependencyResolutionRequiredException e) {
             Log.debug("Failed to resolve dependencies for compile scope, exception: " + e.getMessage());
         }
-        for (final String element : compileClasspathElements) {
+        for (final String element : runtimeClasspathElements) {
             final File f = new File(element);
             try {
                 urlList.add(f.toURI().toURL());

